src/content/3.6/code/agda/snippet03.agda

```diff
@@ -0,0 +1,8 @@
+record Monoid (m : Type 𝑢) : Set (usuc 𝑢) where
+  field
+    unit     : m
+    _⊕_      : m → m → m
+
+    ⊕right-unit : ∀ (x : m) → x ⊕ unit ≡ x
+    ⊕left-unit  : ∀ (x : m) → unit ⊕ x ≡ x
+    ⊕assoc      : ∀ (x y z : m) → (x ⊕ y) ⊕ z ≡ x ⊕ (y ⊕ z)
```

Comment:
You could have predicate [IsMonoid](https://github.com/agda/agda-stdlib/blob/master/src/Algebra/Structures.agda#L160), definition without laws [RawMonoid](https://github.com/agda/agda-stdlib/blob/master/src/Algebra/Bundles/Raw.agda#L38) and combine them to have [Monoid](https://github.com/agda/agda-stdlib/blob/master/src/Algebra/Bundles.agda#L251)

I have chose this encoding as it is:
1. closer to what programmer would expect to have
2. emphasize that in Agda you can actually prove that something is monoid rather than just use property test to check laws as it is done in Scala.